options/posix/include/pthread.h

```diff
@@ -93,6 +93,18 @@ struct  __mlibc_barrier {
 };
 typedef struct __mlibc_barrier pthread_barrier_t;
 
+struct __mlibc_fair_rwlock {
+	pthread_mutex_t m;
+	pthread_mutex_t rw;
```

Comment:
I think we can get away with declaring these members as `unsigned int`. While we do indeed want two mutexes, we do not need the more complex features that pthread_mutex_t supports.

Reply:
I'll modify that